zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -136,114 +131,124 @@ class ZiplineLoader internal constructor(
+  suspend fun loadOnce(
     applicationName: String,
     manifestUrl: String,
+    oldestCodeToLoad: Duration = Duration.INFINITE,
     initializer: (Zipline) -> Unit = {},
   ): Zipline {
-    return try {
-      createZiplineAndLoad(applicationName, manifestUrl, null, initializer)
-    } catch (e: Exception) {
-      createZiplineAndLoad(applicationName, null, null, initializer)
-    }
+    return load(applicationName, flowOf(manifestUrl), oldestCodeToLoad, initializer)
+      .first()
   }
 
   /**
-   * Creates a Zipline for [applicationName] by fetching the manifest, fetching the modules, loading
-   * the code, and running the initializer.
-   *
-   * Pass either a [manifestUrl] or a [providedManifest]. There's no need to pass both.
+   * After identifying a manifest to load this fetches all the code, loads it into a JS runtime,
+   * and runs both the user's initializer and the manifest's specified main function.
    */
-  internal suspend fun createZiplineAndLoad(
+  internal suspend fun loadFromManifest(
     applicationName: String,
-    manifestUrl: String?,
-    providedManifest: LoadedManifest?,
+    loadedManifest: LoadedManifest,
     initializer: (Zipline) -> Unit,
   ): Zipline {
-    eventListener.applicationLoadStart(applicationName, manifestUrl)
     val zipline = Zipline.create(dispatcher, serializersModule, eventListener)
     try {
-      // Load from either pinned in cache or embedded by forcing network failure
-      @Suppress("NAME_SHADOWING")
-      val manifest = providedManifest ?: fetchAndVerifyZiplineManifest(applicationName, manifestUrl)
-      receive(ZiplineLoadReceiver(zipline), manifest, applicationName)
+      receive(ZiplineLoadReceiver(zipline), loadedManifest, applicationName)
 
       // Run caller lambda to validate and initialize the loaded code to confirm it works.
       initializer(zipline)
 
       // Run the application after initializer has been run on Zipline engine.
-      manifest.manifest.mainFunction?.let { mainFunction ->
+      loadedManifest.manifest.mainFunction?.let { mainFunction ->
         zipline.quickJs.evaluate(
-          script = "require('${manifest.manifest.mainModuleId}').$mainFunction()",
+          script = "require('${loadedManifest.manifest.mainModuleId}').$mainFunction()",
           fileName = "ZiplineLoader.kt",
         )
       }
 
-      // Pin stable application manifest after a successful load, and unpin all others.
-      fetchers.pin(applicationName, manifest)
-
-      eventListener.applicationLoadEnd(applicationName, manifestUrl)
       return zipline
     } catch (e: Exception) {
       zipline.close()
-      eventListener.applicationLoadFailed(applicationName, manifestUrl, e)
       throw e
     }
   }
 
-  suspend fun loadContinuously(
-    applicationName: String,
-    manifestUrlFlow: Flow<String>,
-    pollingInterval: Duration,
-    initializer: (Zipline) -> Unit = {},
-  ): Flow<Zipline> = manifestUrlFlow
-    .rebounce(pollingInterval)
```

Comment:
We lost automatic rebounce and polling. Gotta put that back in the development mode code.